op-chain-ops/deployer/opcm/opchain.go

```diff
@@ -30,12 +30,18 @@ type DeployOPChainInput struct {
 	Proposer               common.Address
 	Challenger             common.Address
 
-	BasefeeScalar     uint32
-	BlobBaseFeeScalar uint32
-	L2ChainId         *big.Int
-	OpcmProxy         common.Address
-	SaltMixer         string
-	GasLimit          uint64
+	BasefeeScalar           uint32
+	BlobBaseFeeScalar       uint32
+	L2ChainId               *big.Int
+	OpcmProxy               common.Address
+	SaltMixer               string
+	GasLimit                uint64
+	DisputeGameType         uint32
+	DisputeAbsolutePrestate common.Hash
+	DisputeMaxGameDepth     uint64
+	DisputeSplitDepth       uint64
```

Comment:
Looks like these are uint256's in the dispute game constructor

Reply:
I noticed this and was wondering what way we map to `uint256` in Go. Looks like we use `*big.Int` to represent `uint256`.

Reply:
It would be pretty insane to make the split depth or max game depth larger than a uint64.  Solidity just likes uint256 types a lot.  Restricting these to something sensible would make sense (standard config has them hard coded). A uint32 split depth will give you about 272 years without finalising a valid proposal before you run out of of space. You won't be able to run cannon for a block anywhere near that old in time anyway.

The challenger actually assumes that split depth < 64 (ie number of nodes at split depth +1 fits into a uint64) and that the max depth - split depth <= 64 (the number of nodes within a specific subgame that runs from split depth + 1 to max depth fits into a uint64).

Reply:
I've updated the types for `DisputeMaxGameDepth` and `DisputeSplitDepth` to `uint64` in our Go code, ensuring type safety on the Go side. The task of changing the data types from `uint256` to `uint64` in Solidity should be addressed in a follow-up PR imo.

Reply:
> It would be pretty insane to make the split depth or max game depth larger than a uint64. Solidity just likes uint256 types a lot.

> The task of changing the data types from uint256 to uint64 in Solidity should be addressed in a follow-up PR imo.

@ajsutton @smartcontracts Are you tracking this anywhere? Or, if you are fine with uint256 in the source and just using uint64 in the deploy scripts, that's ok with me and then no need to track it. Just to make sure proofs teams doesn't lose this if you wanted to implement

Reply:
Ah I see @blmalone created https://github.com/ethereum-optimism/optimism/issues/12254, so resolving this thread

Reply:
I'm not unhappy with it staying as uint256 in the code - setting it to the max uint32 is still pretty well shooting yourself in the foot to be honest.  But also not against changing it as it would avoid some confusion about why the type is different between solidity and go.  Just depends how much lift that is given there's limited user value in that change itself.  That's probably a call for @smartcontracts to make and if we don't do it now it can likely be addressed in future work (hopefully split depth won't be needed in the contracts but we'll see...)